packages/models/src/lib/plugin-config-runner.ts

```diff
@@ -24,3 +24,15 @@ export const runnerConfigSchema = z.object(
 );
 
 export type RunnerConfig = z.infer<typeof runnerConfigSchema>;
+
+export const esmObserver = z.object({
+  next: z.function().args(z.unknown()).returns(z.void()).optional(),
+});
+export type EsmObserver = z.infer<typeof esmObserver>;
+
+export const esmRunnerConfigSchema = z
+  .function()
+  .args(esmObserver.optional())
+  .returns(z.promise(auditOutputsSchema));
```

Comment:
I would suggest sync implementation should also be possible.

```suggestion
  .returns(z.union([auditOutputsSchema, z.promise(auditOutputsSchema)]));
```